src/main.cc

```diff
@@ -526,13 +518,12 @@ int main( int argc, char ** argv )
 
   cfg.resetState = gdcl.resetState;
 
-  // Open log file
-  logFilePtr->setFileName( Config::getConfigDir() + "gd_log.txt" );
-  logFilePtr->open( QFile::WriteOnly );
-
-
-  // Install message handler
-  qInstallMessageHandler( gdMessageHandler );
+  // Log to file enabled through command line or preference
+  if ( gdcl.logFile || cfg.preferences.enableApplicationLog ) {
```

Comment:
If user set the option () through the preference , this line will never get the chance to run.

Reply:
Yes?

If the option is not enabled, the default message handler will be working (printing to stdio/console).

If the option is enabled, the custom message handler that prints to gd_log.txt will starts to work.

Reply:
The logFilePtr is useless if the option is not enabled.

Reply:
which means the option in the preference will only work in the next time?

Reply:
Yes? Why not. I think we always want to get full log.

Reply:
Ok, I know what you mean now, and I have decided to move this thing out of main.cc